### PR TITLE
Move MessageContent into Azure.Core

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Addresses https://github.com/Azure/azure-sdk-for-net/issues/27699

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Azure.Messaging.EventHubs.csproj [hunk 1]
```diff
@@ -21,8 +21,7 @@
         <PackageReference Include="Azure.Core" />
     -->
     <!-- END TEMP -->
-    <PackageReference Include="Azure.Core.Experimental" />
-
+    <ProjectReference Include="..\..\..\core\Azure.Core\src\Azure.Core.csproj" />
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Azure.Messaging.EventHubs.csproj:24]: Is this expected to ship as part of the April release of Core?

File: sdk/core/Azure.Core/src/Messaging/MessageContent.cs [hunk 2]
```diff
@@ -5,12 +5,12 @@
 using System.ComponentModel;
 using Azure.Core;
 
-namespace Azure
+namespace Azure.Messaging
 {
     /// <summary>
-    /// Content containing a content type along with its data.
+    /// Message containing a content type along with its data.
```
Comment (jsquire) [inline sdk/core/Azure.Core/src/Messaging/MessageContent.cs:11]:
```suggestion
    /// The content of a message, comprised of the message data and an identifier for the content type.
```

Seems weird to have a class called `MessageContent` and then describe it as a "Message";  feel free to tune the suggestion text as you see fit or ignore it.

**Review by jsquire — COMMENTED** [review]
The changes look good.   My concern is the timing.   If this isn't expected to be part of the Core release for the April milestone, can we delay until after releases are done so that the Event Hubs beta remains bound to the version in Experimental?

**Comment by JoshLove-msft** [comment]
> The changes look good. My concern is the timing. If this isn't expected to be part of the Core release for the April milestone, can we delay until after releases are done so that the Event Hubs beta remains bound to the version in Experimental?

This should be able to go out in the April release.

**Comment by annelo-msft** [comment]
Would it be possible add a description with more context to this PR or the issue?  Where is the API in the proposal -> arch board approval stage?

**Comment by JoshLove-msft** [comment]
> Would it be possible add a description with more context to this PR or the issue? Where is the API in the proposal -> arch board approval stage?

Updated the associated issue - https://github.com/Azure/azure-sdk-for-net/issues/27699

**Review by jsquire — APPROVED** [review]

File: sdk/core/Azure.Core/tests/MessageContentTests.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Messaging;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests
+{
+    public class MessageContentTests
```
Comment (christothes) [inline sdk/core/Azure.Core/tests/MessageContentTests.cs:10]: should we do something with `IsReadOnly` in a test?

**Review by christothes — APPROVED** [review]

**Review by annelo-msft — COMMENTED** [review]
Are we concerned about checking in an unapproved API into Core?  We will need it to be approved at some point, and delaying this could push out Core release dates.

**Comment by JoshLove-msft** [comment]
> Are we concerned about checking in an unapproved API into Core? We will need it to be approved at some point, and delaying this could push out Core release dates.

@tg-msft would you mind approving the Api Review from this PR?

**Comment by JoshLove-msft** [comment]
> > Are we concerned about checking in an unapproved API into Core? We will need it to be approved at some point, and delaying this could push out Core release dates.
> 
> @tg-msft would you mind approving the Api Review from this PR?

I'll send an email to the arch board this change first.

File: sdk/core/Azure.Core/api/Azure.Core.net461.cs [hunk 4]
```diff
@@ -1021,4 +1021,13 @@ public enum CloudEventDataFormat
...
+    public partial class MessageContent
+    {
+        public MessageContent() { }
+        public virtual Azure.Core.ContentType? ContentType { get { throw null; } set { } }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        protected virtual Azure.Core.ContentType? ContentTypeCore { get { throw null; } set { } }
+        public virtual System.BinaryData? Data { get { throw null; } set { } }
+        public virtual bool IsReadOnly { get { throw null; } }
+    }
 }
```
Comment (tg-msft) [inline sdk/core/Azure.Core/api/Azure.Core.net461.cs:1033]: The API change itself looks good to me.

**Comment by JoshLove-msft** [comment]
> > > Are we concerned about checking in an unapproved API into Core? We will need it to be approved at some point, and delaying this could push out Core release dates.
> > 
> > 
> > @tg-msft would you mind approving the Api Review from this PR?
> 
> I'll send an email to the arch board this change first.

Discussed offline - we will merge this in for now and get the approval later this month.

**Pull request merged**